Gems/ROS2/Code/Source/Lidar/ROS2LidarSensorComponent.cpp

```diff
@@ -278,36 +321,39 @@ namespace ROS2
             m_visualisationPoints = m_lastScanResults.m_points;
         }
 
-        const auto inverseLidarTM = entityTransform->GetWorldTM().GetInverse();
-        for (auto& point : m_lastScanResults.m_points)
+        if (!m_canRaycasterPublish)
```

Comment:
you can return early instead 
if (m_canRaycasterPublish)
{ // skip publishing since the data will be published by the raycaster
  return;
}